src/store/modules/order/actions.ts

```diff
@@ -0,0 +1,184 @@
+      if (payload.viewIndex === 0) emitter.emit("dismissLoader");
+    } catch(err) {
+      console.log(err)
+      showToast(translate("Something went wrong"))
+    }
+
+    return resp;
+  },
+
+  async deliverShipment ({ dispatch }, order) {
+    emitter.emit("presentLoader");
+
+    const params = {
+      shipmentId: order.shipmentId,
+      statusId: 'SHIPMENT_SHIPPED'
+    }
+
+    let resp;
+
+    try {
+      resp = await OrderService.updateShipment(params)
+      if (resp.status === 200 && !hasError(resp)) {
+        showToast(translate('Order delivered to', {customerName: order.customerName}))
+      } else {
+        showToast(translate("Something went wrong"))
+      }
+    } catch(err) {
+      console.log(err)
+      showToast(translate("Something went wrong"))
+    }
+
+    emitter.emit("dismissLoader")
+    return resp;
+  },
+
+  async packDeliveryItems ({ commit }, shipmentId) {
+    const params = {
+      shipmentId: shipmentId,
+      statusId: 'SHIPMENT_PACKED'
+    }
+    return await OrderService.updateShipment(params)
+  },
+
+  async quickShipEntireShipGroup ({ dispatch }, payload) {
+    emitter.emit("presentLoader")
+
+    const params = {
+      orderId: payload.order.orderId,
+      setPackedOnly: 'Y',
+      dimensionUomId: 'WT_kg',
+      shipmentBoxTypeId: 'YOURPACKNG',
+      weight: '1',
+      weightUomId: 'WT_kg',
+      facilityId: payload.facilityId,
+      shipGroupSeqId: payload.shipGroup
+    }
+    
+    let resp;
+
+    try {
+      resp = await OrderService.quickShipEntireShipGroup(params)
+      if (resp.status === 200 && !hasError(resp) && resp.data._EVENT_MESSAGE_) {
+        /* To display the button label as per the shipmentMethodTypeId, this will only used on orders segment.
+          Because we get the shipmentMethodTypeId on items level in wms-orders API.
+          As we already get shipmentMethodTypeId on order level in readytoshiporders API hence we will not use this method on packed orders segment.
+        */
+        const shipmentMethod = payload.order.items.find((ele: any) => ele.shipGroupSeqId == payload.shipGroup).shipmentMethodTypeId
+        if (shipmentMethod !== 'STOREPICKUP') {
+          const shipmentId = resp.data._EVENT_MESSAGE_.match(/\d+/g)[0]
+          await dispatch('packDeliveryItems', shipmentId).then((data) => {
+            if (!hasError(data) && !data.data._EVENT_MESSAGE_) showToast(translate("Something went wrong"))
+          })
+        }
+        showToast(translate("Order packed and ready for delivery"))
+      } else {
+        showToast(translate("Something went wrong"))
+      }
+    } catch(err) {
+      console.log(err)
+      showToast(translate("Something went wrong"))
+    }
+
+    emitter.emit("dismissLoader")
+    return resp;
+  },
+
+  // TODO: handle the unfillable items count
+  async setUnfillableOrderOrItem ({ dispatch }, order) {
+    emitter.emit("presentLoader");
+    return await dispatch("rejectOrderItems", order).then((resp) => {
```

Comment:
Why 2 different methods?
Why we are dispatching action of same state?
Is it possible to have partial rejection of items or all items will be rejected?

Reply:
Sir, using two different methods to separate the service call and handle the promises.
>> Is it possible to have partial rejection of items or all items will be rejected?
Sir, for now, all items will be rejected.